packages/launchpad/src/signer.ts

```diff
@@ -0,0 +1,38 @@
+import { StdSignDoc } from "./encoding";
+import { StdSignature } from "./types";
+
+export type Algo = "secp256k1" | "ed25519" | "sr25519";
+
+export interface AccountData {
+  /** A printable address (typically bech32 encoded) */
+  readonly address: string;
+  readonly algo: Algo;
+  readonly pubkey: Uint8Array;
+}
+
+export interface SignResponse {
+  /**
+   * The sign doc that was signed.
+   * This may be different from the input signDoc when the signer modifies it as part of the signing process.
+   */
+  readonly signed: StdSignDoc;
+  readonly signature: StdSignature;
+}
+
+export interface OfflineSigner {
+  /**
+   * Get AccountData array from wallet. Rejects if not enabled.
+   */
+  readonly getAccounts: () => Promise<readonly AccountData[]>;
+
+  /**
+   * Request signature from whichever key corresponds to provided bech32-encoded address. Rejects if not enabled.
+   *
+   * The signer implementation may offer the user the ability to override parts of the signDoc. It must
+   * return the doc that was signed in the response.
+   *
+   * @param signerAddress The address of the account that should sign the transaction
+   * @param signDoc The content that should be signed
+   */
+  readonly sign: (signerAddress: string, signDoc: StdSignDoc) => Promise<SignResponse>;
```

Comment:
@Thunnini can you have a look in this file? This allows the signer implementation to override any field in the SignDoc. The flow is:

1. Application sends a `StdSignDoc` to be signed
2. The signer shows it tot he and (optionally) allows the user to modify it
3. User approves
4. Signer serializes `StdSignDoc` (e.g. using `serializeSignDoc`) and creates a signature
5. Signer returns the signature and the final document it signed at the end of the day
6. Application can now double check what was signed (or just trust the signer) and broadcast

This should cover most of your requested flexibility, just in one call instead of multiple calls.